src/api/mapping.ts

```diff
@@ -23,13 +23,13 @@ mappingRouter.post('/', async (req, res) => {
   if (parseInt(req.body.port) < 3001) {
     return res.status(400).json({ message: 'Port cannot be smaller than 3001' })
   }
-  const existingSubDomain = domainKeys.find(
-    e => e.subDomain === req.body.subDomain
-  )
+  const existingSubDomain =
+    domainKeys.find(e => e.subDomain === req.body.subDomain) &&
+    domainKeys.find(e => e.domain === req.body.domain)
```

Comment:
since you discovered a bug, could you add a test to make sure nobody regresses on thhis bug inn the future?

Reply:
Ideally, this bug fix and your `status` feature, should have been 2 separate PRs. In this case, since its already done, its okay. Would need a test for this fix, thank you.

Reply:
what about finding `fullDomains` instead.... does that work?

Reply:
I wrote the test for it. Let me see if i can figure out how to do it with one check.